src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -146,7 +146,33 @@ public IList<ITelemetryInitializer> TelemetryInitializers
         {
             get { return this.telemetryInitializers; }
         }
-        
+
+        /// <summary>
+        /// Gets or sets the first telemetry processor in the chain of processors.
+        /// </summary>
+        public TelemetryProcessorChain TelemetryProcessorChain
+        {
+            get
+            {                
+                if (this.telemetryProcessorChain == null)
+                {
+                    new TelemetryProcessorChainBuilder().Build(this);
+                }
+
+                return this.telemetryProcessorChain;
+            }
+
+            set
```

Comment:
Yes, we need public getter. Public setter makes you check whether it's null or not every time you access it. It is a minimal problem I can think of.